code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -1,13 +1,24 @@
-//The mob should have a gender you want before running this proc. Will run fine without H
+/datum/preferences/proc/random_hair_style()
+	var/decl/species/mob_species = get_species_by_key(species || global.using_map.default_species)
+	var/decl/bodytype/mob_bodytype = mob_species?.get_bodytype_by_name(bodytype) || mob_species?.default_bodytype || GET_DECL(global.using_map.default_bodytype)
```

Comment:
```suggestion
	var/decl/bodytype/mob_bodytype = get_bodytype_decl()
```
`using_map.default_bodytype` is deprecated as far as I know. IMO it should be removed and all uses of it replaced with the default species' default bodytype.